wgpu-core/src/device/mod.rs

```diff
@@ -699,6 +694,7 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                     };
                 }
                 Err(e) => {
+                    // TODO: return error on failure?
                     log::error!("failed to create buffer in a mapped state: {:?}", e);
```

Comment:
I cannot think of a scenario when we couldn't map a buffer at creation time and then it would work later. So I'm thinking it's better to return an error here :thinking:?

Reply:
if it can't happen, let's just assert on that